device/device.go

```diff
@@ -396,8 +396,11 @@ func (d *Device) startStreamLoop(ctx context.Context) error {
 		var frame []byte
 		ioMemType := d.MemIOType()
 		bufType := d.BufferType()
-		waitForRead := v4l2.WaitForRead(d)
-		for {
+		doneChan := make(chan struct{})
+		waitForRead := v4l2.WaitForRead(d, doneChan)
+
+		defer close(doneChan)
+		for d.streaming {
```

Comment:
@doertydoerk, this change will introduce a read-race condition for `d.streaming` and adding a mutex lock will definitely impact performance. 

`d.streaming` is not used for signaling, but for state update.  So I think, may be a different approach may work better where the context is used to signal closing when `device.Close` is called.

First, update type `Device` to store to store the `Context.CancelFunc` value
```go
type Device struct {
    ...
    startCtx context.Context
    startCtxCancelFn context.CancelFunc
}
```

Then, in `Device.Start` extract and store the cancel function
```go
func (d *Device) Start(ctx context.Context) error {
    ...
    d.startCtx, d.startCtxCancelFunc = context.WithCancel(ctx)
    ...
    if err := d.startStreamLoop(d.startCtx); err != nil {
        return fmt.Errorf("device: start stream loop: %s", err)
    }
}
```

Now, the `Device.Close` can use the start context to safely signal `Done`

```go
func (d *Device) Stop() error {
    if !d.streaming {
        return nil
    }
    ...
    d.startCtxCancelFunc() // signals ctx.Done to streaming loop.
}
```

Let me know what you think.

Reply:
@doertydoerk Also, you mentioned the `go routine kept building up real fast` can you elaborate on that ? I did have a memory leak issue in previous version, which was fixed here https://github.com/vladimirvivien/go4vl/issues/22. Let me know if you are still seeing this behavior in the latest released version.

Reply:
Hi @vladimirvivien, thanks for reviewing. Couple of fair points there. Will look into it as soon as I get a chance. 
You may disregard the issue of the fast building goroutines. We ran into that but as you indicated, that was already resolved by updating to the latest version. Little confusion on my part 😉